### PR TITLE
[Dashboard] Display actor task execution info

### PR DESCRIPTION
**Pull request by zzyunzhi**
## Why are these changes needed?

Display number of executed tasks, actor creation/reconstruction timestamp, approximate task execution speed. 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: src/ray/raylet/node_manager.cc [hunk 1]
```diff
@@ -3050,6 +3051,36 @@ void NodeManager::HandleNodeStatsRequest(const rpc::GetNodeStatsRequest &request
   }
 }
 
+void NodeManager::HandleProfilingStatsRequest(const rpc::GetProfilingStatsRequest &request,
```
Comment (ashione) [inline src/ray/raylet/node_manager.cc]: Why not handle this stats request in job agent or some plugin like it?
It's heavy overhead in raylet/nodemanager.

File: python/ray/reporter.py [hunk 2]
```diff
@@ -157,6 +185,14 @@ def perform_iteration(self):
...
 
     def run(self):
+        """Publish the port."""
+        thread_pool = futures.ThreadPoolExecutor(max_workers=10)
+        server = grpc.server(thread_pool, options=(("grpc.so_reuseport", 0), ))
+        reporter_pb2_grpc.add_ReporterServiceServicer_to_server(
+            ReporterServer(), server)
+        port = server.add_insecure_port("[::]:0")
+        server.start()
+        self.redis_client.set(self.ip, port)
```
Comment (pcmoritz) [inline python/ray/reporter.py]: This should be replaced by something like `self.redis_client.set("REPORTER_PORT:" + self.ip, port)` and then similar for the `.get` call above, to make sure the key is unique and doesn't clash with anything else.

File: python/ray/dashboard/dashboard.py [hunk 3]
```diff
@@ -101,6 +104,7 @@ def __init__(self,
                  temp_dir,
                  redis_password=None):
         """Initialize the dashboard object."""
+        print(redis_address, redis_password)
```
Comment (pcmoritz) [inline python/ray/dashboard/dashboard.py]: we should get rid of this

File: src/ray/protobuf/reporter.proto [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+message GetProfilingStatsReply {
+  // Profiling stats.
+  string profiling_stats = 1;
+  string stdout = 2;
+  string stderr = 3;
+}
+
+// Service for inter-node-manager communication.
```
Comment (pcmoritz) [inline src/ray/protobuf/reporter.proto]: This should be `Service for communicating with the reporter.py process on a remote node`

File: src/ray/protobuf/reporter.proto [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  // PID of the worker process.
+  uint32 pid = 1;
+  // Duration of the profiling in seconds.
+  int32 duration = 2;
+}
+
+message GetProfilingStatsReply {
+  // Profiling stats.
+  string profiling_stats = 1;
+  string stdout = 2;
```
Comment (pcmoritz) [inline src/ray/protobuf/reporter.proto]: Comment `// Standard output of the profiler` and `// Standard error of the profiler` below

File: src/ray/core_worker/core_worker.cc [hunk 6]
```diff
@@ -1189,6 +1189,8 @@ void CoreWorker::HandleGetCoreWorkerStats(const rpc::GetCoreWorkerStatsRequest &
     (*used_resources_map)[it.first] = quantity;
   }
   stats->set_webui_display(webui_display_);
+  RAY_LOG(WARNING) << actor_title_;
```
Comment (pcmoritz) [inline src/ray/core_worker/core_worker.cc]: This should be removed

File: src/ray/raylet/node_manager.cc (hunk 1 above)
Comment (pcmoritz) [reply src/ray/raylet/node_manager.cc]: This is now fixed btw (it goes through the reporter, which is off the critical path). Thanks for the suggestion :)

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**